app/custom_domain_validation.py

```diff
@@ -0,0 +1,33 @@
+from app.db import Session
+from app.dns_utils import get_cname_record
+from app.models import CustomDomain
+
+
+class CustomDomainValidation:
+    def __init__(self, dkim_domain: str):
+        self.dkim_domain = dkim_domain
+
+    def get_dkim_records(self, custom_domain: CustomDomain) -> {str: str}:
+        """
+        Get a list of dkim records to set up. It will be
+
+        """
+        return {
+            (f"{key}._domainkey", f"{key}._domainkey.{self.dkim_domain}")
+            for key in ("dkim", "dkim02", "dkim03")
+        }
+
+    def validate_dkim_records(self, custom_domain: CustomDomain) -> {str: str}:
+        """
+        Check if dkim records are properly set for this custom domain.
+        Returns empty list if all records are ok. Otherwise return the records that aren't properly configured
+        """
+        invalid_records = {}
+        for prefix, expected_record in self.get_dkim_records(custom_domain):
+            custom_record = f"{prefix}.{custom_domain.domain}"
+            dkim_record = get_cname_record(custom_record)
+            if dkim_record != expected_record:
+                invalid_records[custom_record] = dkim_record or "empty"
+        custom_domain.verified = len(invalid_records) == 0
+        Session.commit()
```

Comment:
I think we shouldn't invalidate DKIM right away as if user sets a enforced DMARC, emails from their aliases will fail DMARC. We should rather just display a warning and keep the `dkim_verified` unchanged.

Reply:
Hm.. it could happen now couldn't it? If somebody sets DMARC but not DKIM they'll end up in the same situation. 

We could disable dkim if they don't have DMARC. Otherwise show a warning.

Reply:
The issue is for users who have current DKIM (with 1 CNAME) correctly set up and have DKIM enforced in their DMARC. Let's say we send an email to let them know about the change, they go the DNS page, validate their domain again and see the DKIM error but don't have time to make the change or maybe their DNS provider takes a while to propagate the change. In both cases, emails sent from their alias in the meantime won't have DKIM signature added and will be rejected because of their DMARC policy.

Reply:
Yep, This is what I propose: 
 - User has dkim enabled with one cname and dmarc
 - Validates dkim, fails because now we require 3
 - If DMARC is validated, we don't disable dkim but show a warning
 - if DMARC is not validated we disable dkim and show the warning

Reply:
To be on the safe side, I think we shouldn't disable DKIM if it's currently enabled. User can have their own DMARC setting (to have DMARC reporting for ex) and we don't know that. We'll leave a bit of time (1-3 months) for users to update DKIM and after that, we can use a stricter policy.

Reply:
Ok.